### PR TITLE
Add type synonym 'InterpreterFor' and documentation

### PR DESCRIPTION
**Pull request by bolt12**
Hi,

This PR tackles issue #220 . I've put the type synonym on the Polysemy.Internal module and documented it.

Please let me know if I should improve anything.

### CODE REVIEW

File: src/Polysemy/Internal.hs [hunk 1]
```diff
@@ -434,6 +435,18 @@ runM (Sem m) = m $ \z ->
...
+-- | Type synonym for interpreters that consume an effect without changing the 
+-- return value. Offered for user convenience. 
+--
+-- 'r' Is kept polymorphic so it's possible to place constraints upon it:
+--
+-- @
+-- teletypeToIO
+-- :: 'Member' (Embed IO) r
+-- => 'InterpreterFor' Teletype r
+-- @
```
Comment (KingoftheHomeless) [inline src/Polysemy/Internal.hs]: I _think_ you need another comment line here to make the code block render properly. I don't quite get how Haddock works on that point. Also, the indentation is a bit wrong here.

To illustrate what I mean, here's how I'd do it:

```haskell
-- @r@ Is kept polymorphic so it's possible to place constraints upon it:
--
-- @
-- teletypeToIO
--   :: 'Member' (Embed IO) r
--   => 'InterpreterFor' Teletype r
-- @
--
```

File: src/Polysemy.hs [hunk 2]
```diff
@@ -14,6 +14,10 @@ module Polysemy
   , Embed (..)
   , embed
   , embedToFinal
+
+  -- * Type synonyms for user convenience
+  -- ** InterpreterFor
+  , InterpreterFor
```
Comment (KingoftheHomeless) [inline src/Polysemy.hs]: This is exported in the wrong place; in the middle of `Interoperating with other monads`, which will make the documentation look weird. Instead, place this after the export of `runFinal`.

Also, you don't need the `InterpreterFor` subheading.

File: src/Polysemy.hs [hunk 3]
```diff
@@ -132,11 +136,11 @@ module Polysemy
...
-import Polysemy.Internal.TH.Effect
-import Polysemy.Internal.Tactics
-import Polysemy.Final
+import           Polysemy.Final
+import           Polysemy.Internal
+import           Polysemy.Internal.Combinators
+import           Polysemy.Internal.Forklift
+import           Polysemy.Internal.Kind
+import           Polysemy.Internal.Tactics
+import           Polysemy.Internal.TH.Effect
```
Comment (KingoftheHomeless) [inline src/Polysemy.hs]: I'd say the extra indentation isn't needed here; I doubt we'll ever use quantified modules in `Polysemy`. The reordering is good, though.

Comment (bolt12) [reply src/Polysemy.hs]: Oops! My setup indents imports by default, I forgot to turn this off. I will revert this changes!

File: src/Polysemy.hs (hunk 2 above)
Comment (bolt12) [reply src/Polysemy.hs]: Okay Thanks! I wasn't sure where to export it. I will correct this!

File: src/Polysemy/Internal.hs (hunk 1 above)
Comment (bolt12) [reply src/Polysemy/Internal.hs]: I have generated the docs and everything looked alright! I'll make this change and see how it looks!

**Comment by KingoftheHomeless** [comment]
@isovector When it comes to simple pull requests like these, do you also want to take a look at them, or can I merge them at my discretion?

**Comment by isovector** [comment]
@KingoftheHomeless i trust you! thanks for taking the initiative on this!

**Comment by KingoftheHomeless** [comment]
Thanks for your work @bolt12!

**Comment by bolt12** [comment]
@KingoftheHomeless No problem! My pleasure, I hope to continue contributing :)

**Pull request merged**